src/lib/stepper/stepper-intl.ts

```diff
@@ -22,3 +22,16 @@ export class MatStepperIntl {
   /** Label that is rendered below optional steps. */
   optionalLabel: string = 'Optional';
 }
+
+
+/** @docs-private */
+export function MAT_STEPPER_INTL_PROVIDER_FACTORY(parentIntl: MatStepperIntl) {
+  return parentIntl || new MatStepperIntl();
```

Comment:
I'm not sure we want to do this; it makes it impossible to re-provide the intl config for a subset of the app. I would have thought that including the `provide:` for the custom implementation _in_ the lazy-loaded module would bring in the customization.

Reply:
It doesn't look like that's the case, based on our discussion in the issue. FWIW we do the same as here for the sort header and paginator.

Reply:
In the issue it said that they were providing their custom intl implementation in the root module and not in the lazy-loaded module, which I think is the problem. Using `providedIn` isn't an option here because they have to re-provide the existing token.

Reply:
That means that they'd have to remember to add the provider for each new lazy-loaded module that is introduced. I don't think that a translation provider is something you'd generally want to be able to re-provide for different sections of the app. Also if we went with the `providedIn` approach, how would we handle the other similar providers that already use the old approach?

Reply:
I don't follow what you mean in the second part about `providedIn`

Reply:
I mean that if we went with the approach that is currently in master (having a `providedIn` and having people provide it themselves in the lazy-loaded modules), it makes the stepper's i18n provider inconsistent with the other i18n providers (like sort header or paginator) which are provided in the same way as this PR.